sphinxcontrib/needs/directives/needpie.py

```diff
@@ -156,10 +154,16 @@ def process_needpie(app, doctree, fromdocname):
                 if filter_func:
                     filter_func(**context)
                 sizes = context["results"]
+                # check items in sizes
+                for item in sizes:
+                    if not isinstance(item, int) and not isinstance(item, float):
+                        logger.error(f"{item} in sizes {sizes} is not a integer or float number.")
```

Comment:
"sizes" is an internal name from the code, the user does not know it.
So the error message is not really helpful.
Better : The returned values from the given filter_func is not valid. It must be a list with items of type int/float.